src/services/expences.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+class Expences {
+  constructor() {
+    this.expenses = [];
+  }
+
+  reset() {
+    this.expenses = [];
+  }
+
+  getAll(userId, category, from, to) {
+    const filterByUserId = (expense) => {
+      return !userId || expense.userId === Number(userId);
+    };
+
+    const filterByCategory = (expense) => {
+      return !category || category.includes(expense.category);
+    };
+
+    const filterByTime = (expense) => {
+      const expenseDate = new Date(expense.spentAt);
+      const fromDate = from ? new Date(from) : null;
+      const toDate = to ? new Date(to) : null;
+
+      return (!fromDate || expenseDate >= fromDate)
+        && (!toDate || expenseDate <= toDate);
+    };
+
+    this.expenses = this.expenses
+      .filter(filterByUserId)
+      .filter(filterByCategory)
+      .filter(filterByTime);
+
+    return this.expenses;
+  }
+
+  getById(expenceId) {
+    const foundExpense = this.expenses
+      .find(expense => Number(expenceId) === expense.id);
+
+    return foundExpense || null;
+  }
+
+  createExpence(data) {
+    const newUser = {
+      id: this.expenses.length + 1,
```

Comment:
Instead of using the users' length to generate the new user ID, use a variable to store the next available ID and increment it every time a new user is created. This will avoid potential ID conflicts when users are removed.
Make a function for it or use uuid lib for it

Reply:
if i clearly understand tests require number value in id 
![image](https://github.com/mate-academy/node_accounting-app-v2/assets/124785462/a63de23e-3aac-4070-a94c-896130b01847)